compiler/rustc_mir_transform/src/add_call_guards.rs

```diff
@@ -50,10 +50,10 @@ impl AddCallGuards {
         for block in body.basic_blocks_mut() {
             match block.terminator {
                 Some(Terminator {
-                    kind: TerminatorKind::Call { target: Some(ref mut destination), cleanup, .. },
+                    kind: TerminatorKind::Call { target: Some(ref mut destination), unwind, .. },
                     source_info,
                 }) if pred_count[*destination] > 1
-                    && (cleanup.is_some() || self == &AllCallEdges) =>
+                    && (matches!(unwind, UnwindAction::Cleanup(_)) || self == &AllCallEdges) =>
```

Comment:
Yeah, already independently discovered this. Fix pushed.

Reply:
The old logic is actually wrong already, because `None` can also map to `invoke` and calls terminate in cleanup blocks, but since we only generate drop calls in cleanup blocks and these return `()`, we never actually hit the issue.